### ISSUE TITLE
Ios recorder audio is m4a and android recorded is mp4

### ISSUE THREAD

**Issue by DevHamzaShahid** [hyochan/react-native-audio-recorder-player#588]
> Please fill the template to help you out. Also, please try the `Example` project compare before submiting the issue when you have certain issue with your project setup.

### Version of react-native-audio-recorder-player
"react-native-audio-recorder-player": "^3.6.6",

### Version of React Native"
"react-native": "0.73.1",

### Platforms you faced the error (IOS or Android or both?)
in android
### Expected behavior
Ios recorded audio is m4a which should also play in android
### Actual behavior
android recorded audio is playing in both platforms as it is recorded with extension mp4
but ios recorded m4a audio is only playable on ios not when we play that audio in android
as i am working on chat app, so ios users are sending audio chat to android user then android users are unable to play

**Comment by yernandus** [comment]
same thing. android is recording mp4

**Comment by LiuYiSong** [comment]
Yes, this is a very big problem. If Android and iOS files cannot be played together, it will be meaningless, so it is urgent to solve this problem

**Comment by TallNutAlt** [comment]
一样

**Comment by TallNutAlt** [comment]
the same

**Comment by Irfan-S** [comment]
By saving files as .AAC on both iOS and Android, this issue can be worked around.

`const audioSet: AudioSet = {
  AudioEncoderAndroid: AudioEncoderAndroidType.AAC,
  AudioSourceAndroid: AudioSourceAndroidType.MIC,
  AVEncoderAudioQualityKeyIOS: AVEncoderAudioQualityIOSType.high,
  AVNumberOfChannelsKeyIOS: 2,
  AVFormatIDKeyIOS: AVEncodingOption.aac,
};
`
This can then be passed in as a prop during recording, which then works on both devices

**Comment by TallNutAlt** [comment]
> 通过在 iOS 和 Android 上将文件保存为 .AAC，可以解决此问题。
> 
> ```ts
> const audioSet: AudioSet = {
>   AudioEncoderAndroid: AudioEncoderAndroidType.AAC,
>   AudioSourceAndroid: AudioSourceAndroidType.MIC,
>   AVEncoderAudioQualityKeyIOS: AVEncoderAudioQualityIOSType.high,
>   AVNumberOfChannelsKeyIOS: 2,
>   AVFormatIDKeyIOS: AVEncodingOption.aac,
> };
> ```
> 
> 然后可以在录制期间将其作为道具传入，然后在两个设备上都可以使用

const audioSet = {
      AudioEncoderAndroid: AudioEncoderAndroidType.AAC,
      AudioSourceAndroid: AudioSourceAndroidType.MIC,
      AVEncoderAudioQualityKeyIOS: AVEncoderAudioQualityIOSType.high,
      AVNumberOfChannelsKeyIOS: 2,
      AVFormatIDKeyIOS: AVEncodingOption.aac,
    };
    audioRecorderPlayer.startRecorder(undefined,audioSet).then(res => {
      console.log(res);
      setNextYuyin(true);
    });
    我尝试了一下 并没有变成 aac格式

**Comment by skontan** [comment]
I have the same problem. I can't play android recorded audio (.mp4) on my ios device

**Comment by AdityaThummar** [comment]
+1

**Comment by hyochan** [comment]
Hi — closing this as `🗑 stale-legacy`. This issue references the pre-Nitro `react-native-audio-recorder-player` API, which has since been rewritten on top of [react-native-nitro-modules](https://github.com/mrousavy/nitro) and republished as `react-native-nitro-sound`. The internals, method names, and error surfaces are different.

If this still reproduces on the **current `react-native-nitro-sound`**, please open a fresh issue with:
- `react-native-nitro-sound` version
- React Native version
- Platform (iOS/Android) + OS version
- A minimal reproduction

We'll triage it promptly. Thanks for your patience. 🙏

> 🤖 This repo is now AI-native maintained — see [#787](https://github.com/hyochan/react-native-nitro-sound/discussions/787).
